### ISSUE TITLE
Add package with existing name results in package in two projects

### ISSUE THREAD

**Issue by LisaAlmarode** [ericwinger/Jade#266]
If you have a package named Test in one project, and then try to add a new package named Test to a different project, it results in the same package being in both projects.  This follows from how package and project definitions work.

However, if you unload the second-of the projects, it unloads the package from the first and leaves the second project still present but marked dirty.  Weird stuff happens.

**Comment by ericwinger** [comment]
Hmm, I don't see "weird stuff." Nor do I see the removal from the first package in Rowan if the second project is unloaded. Maybe I'm misunderstanding. 

```
(Rowan image loadedProjectNamed: 'Rowan') loadedPackages detect:[:package | package name = 'Foo'] aRwGsLoadedSymbolDictPackage Foo

(Rowan image loadedProjectNamed: 'RowanSample1') loadedPackages detect:[:package | package name = 'Foo'] aRwGsLoadedSymbolDictPackage Foo

<unload project RowanSample1 in Jadeite>

(Rowan image loadedProjectNamed: 'Rowan') loadedPackages detect:[:package | package name = 'Foo']  aRwGsLoadedSymbolDictPackage Foo
```

**Comment by LisaAlmarode** [comment]
With VegTest project loaded, I add a package named Vegetables to RowanSample3.  It's now in two places:
![image](https://user-images.githubusercontent.com/13142166/48579362-1d2b8380-e8d1-11e8-92cc-6c383e90dc05.png)

Then I unload RowanSample3.  This unloads it from VegTest:
![image](https://user-images.githubusercontent.com/13142166/48579421-44825080-e8d1-11e8-86ad-777e867abc06.png)

**Comment by ericwinger** [comment]
@dalehenrich   It is my faint recollection that Rowan can't yet ( or never will ) treat package names independently between projects. i.e. A package name is canonical throughout the entire system. 
Is that correct? 

@LisaAlmarode - If so, that would explain this behavior. When Rowan was asked to add a "new" package, it retrieved the existing package.

**Comment by dalehenrich** [comment]
... I haven't looked at the details here, but I'm curious what behavior you will want> I can throw an error if you try to add an existing package, but I assume that that would be harsh for the GUI to deal with -- is this a good error (i.e. the user needs to be informed that the package already exists) or a bad error (i.e., you've hit a bug in Rowan) ... Jadeite could check to see if it is an existing package and inform the user directly or Rowan could check to see if it an existing package and then ??? (signal an inform request or ...) 

My inclination is to make it an error to try to add an existing package, which means Jadeite will need to check whether or not the package exists before trying to add the package, so the GUI can interact rationally with the user ... assigning back to @ericwinger and opening a Rowan issue to raise an error in this case ...

**Comment by ericwinger** [comment]
I'd be ok with Jadeite preventing the addition of a duplicate package regardless of project. @LisaAlmarode ?

**Comment by LisaAlmarode** [comment]
Yes, I'm fine with disallowing the duplicate add.

**Comment by ericwinger** [comment]
Duplicate packages prevented in Jadeite. Warning messages if user tries. Fix will be in Oscar 3.0.53.

**Comment by LisaAlmarode** [comment]
Problem avoided.  I think it's going to be an issue that we can't find the location of the blocking package, but that can be a separate issue.
